packages/components/src/custom-select-control-v2/index.tsx

```diff
@@ -1,5 +1,9 @@
 /**
  * Internal dependencies
  */
-export { default as CustomSelect } from './default-component';
-export { default as CustomSelectItem } from './custom-select-item';
+import CustomSelect from './default-component';
+import Item from './item';
+
+const CustomSelectControlV2 = Object.assign( CustomSelect, { Item } );
```

Comment:
Do we really need to use `Object.assign()` rather than just do a direct assignment?

Reply:
It is necessary for TypeScript reasons because here the component is already exported once and "sealed" to new keys. 

But I moved the assignment to the default-component file so it's a bit simpler and doesn't require the `Object.assign` (7b678d4097ffff5824d6c8b2bf78205ed8ac64fe).